grommet/grommet
# Improve Layer + Target scroll behavior

#### What does this PR do?
Updates the way layer is styled to move with a target to remove scroll lag. Previously, Layer bounds were being handled by a state variable, so every time a user scrolled or resized the browser, a state variable was being updated and then affecting the CSS. This was a costly computation which caused the movement of the Layer to lag behind the movement of the target.

These updates align the behavior of Layer positioning closer to how Drop behaves (where no scroll lag is exhibited). Mainly, instead of relying on a state variable, the layer container style is now adjusted directly in the scroll/resize listener functions.

Some other small changes to take note of:
1. When Layer (modal or non-modal) has a target, it should be positioned `absolute` to allow it to align properly and move with its target.
2. A CSS property of `will-change: transform` is added to LayerOverlay in order to improve scroll performance. Prior, the overlay was being repainted on every render which also could have been contributing to some lag. This CSS property enhances the way the browser renders this element so it is not repainted on scroll.

Snapshot changes are merely additions of the `will-change` property and addition of the Layer wrapper to Layers with `modal={false} + target`.

#### Where should the reviewer start?
src/js/components/Layer/LayerContainer.js

#### What testing has been done on this PR?
Tested scroll behavior with: https://storybook.grommet.io/?path=/story/layout-layer--target-layer
and checked other Layer stories to ensure no changed behavior.

#### How should this be manually tested?
Storybook deploy of "Layer Target" story. Open layer, (modal or non-modal), and scroll. No lag should be seen.

#### Any background context you want to provide?

#### What are the relevant issues?
Related to https://github.com/grommet/hpe-design-system/issues/1115

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Enhanced Layer + `target` scroll behavior to remove scroll lag

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
